### ISSUE TITLE
Trigger subscription callback when there is a publish action

### ISSUE THREAD

**Issue by flyfy1** [palkan/action-cable-testing#24]
When using [GraphqlRuby](https://github.com/rmosolgo/graphql-ruby/), I'm using ActionCable to support Graphql#Subscription. In order to support that, I need to write a customized Channel for the updates.

Therefore, I'm trying to write test for my Channel implementation, using this library.

I'm writing code to test the general channel behaviour, that works fine; however, when I'm trying to test the flow of "server push data to client", I'm facing a problem with the incompatibility between how Graphql#Subscription is implemented by GraphqlRuby, and this library:

## How Graphql#Subscription is implemented by GraphqlRuby

When there is a subscription request, sth like:

```graphql
    subscription SubscribeUser($uuid: String) {
      user(uuid: $uuid) {
        name
        avatar {
          cloudinaryId
          cloudinaryType
        }
      }
    }
```

2 subscriptions would happen:

1. it would subscribe for any updates, based on the query param: <https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/subscriptions/action_cable_subscriptions.rb#L101>; and furthermore, executes the query (i.e., trigger next step)
2. it would subscribe for a given subscription_id, and whenever a broadcast happens (from first step), it would deliver this broadcast to the corresponding client
<https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/subscriptions/action_cable_subscriptions.rb#L97>, 

In short, this library is using ActionCable#pubsub in 2 ways:
1. the normal way as we're expecting, push directly to client
2. using PubSub as a way for passing messages internally

## The problem I'm facing

It seems to me that `action-cable-testing` has overwrote the Connection & Channel: <https://github.com/palkan/action-cable-testing/blob/master/lib/action_cable/channel/test_case.rb#L21>, therefore making it unable to trigger the default PubSub callback (i.e., step 2 above).

So I want to ask, in such a case, what do u recommend for testing part of triggering callback internally?

**Comment by palkan** [comment]
I think, it's possible to add support for _real_ `broadcast` in unit-tests:
- implement `subscribe` / `unsubscribe` for `Test` adapter
- enhance `stream_from` and `stop_all_streams` for `ChannelStub` to interact with pubsub.

I'm not sure when I'll have time to do that by myself. If you'd like to propose a PR, feel free to ask me for advice/help!

**Comment by palkan** [comment]
Hi @flyfy1!

Could you, please, provide an example of a channel and the corresponding test which involves streams callbacks?

Having a good example would help to figure our a better way to implement support for this.

**Comment by flyfy1** [comment]
@palkan Sorry it has been a while since last time I was working on this issue 😂

I've changed the way I test things to bypass this issue now... would probably change it back later on (at that time, I'll re-open this issue)
